src/Compilers/CSharp/Portable/Parser/SlidingTextWindow.cs

```diff
@@ -34,207 +45,165 @@ internal sealed class SlidingTextWindow : IDisposable
+        /// + <see cref="_characterWindow"/>'s <see cref="ArraySegment{T}.Count"/>.
         /// </summary>
-        public int LexemeRelativeStart
-        {
-            get
-            {
-                return _lexemeStart;
-            }
-        }
+        private int CharacterWindowEndPositionInText => _characterWindowStartPositionInText + _characterWindow.Count;
 
         /// <summary>
-        /// Number of characters in the character window.
+        /// Returns true if <paramref name="position"/> is within the current character window, and thus the character at that position
+        /// can be read from the character window without going back to the underlying <see cref="Text"/>.
         /// </summary>
-        public int CharacterWindowCount
+        private bool PositionIsWithinWindow(int position)
         {
-            get
-            {
-                return _characterWindowCount;
-            }
+            return position >= _characterWindowStartPositionInText &&
+                   position < this.CharacterWindowEndPositionInText;
         }
 
         /// <summary>
-        /// The absolute position of the start of the current lexeme in the given
-        /// SourceText.
+        /// Returns true if <paramref name="span"/> is within the current character window, and thus the sub-string corresponding to 
+        /// that span can be read can be read from the character window without going back to the underlying <see cref="Text"/>.
         /// </summary>
-        public int LexemeStartPosition
+        public bool SpanIsWithinWindow(TextSpan span)
         {
-            get
-            {
-                return _basis + _lexemeStart;
-            }
+            return span.Start >= _characterWindowStartPositionInText &&
+                   span.End <= this.CharacterWindowEndPositionInText;
         }
 
         /// <summary>
-        /// The number of characters in the current lexeme.
+        /// Returns the span of characters corresponding to <paramref name="span"/> from the character window.
+        /// The <paramref name="span"/> must be within the bounds of the current character window (see
+        /// <see cref="SpanIsWithinWindow(TextSpan)"/>).
         /// </summary>
-        public int Width
+        public ReadOnlySpan<char> GetTextOfValidSpan(TextSpan span)
         {
-            get
-            {
-                return _offset - _lexemeStart;
-            }
+            Debug.Assert(SpanIsWithinWindow(span));
+            return _characterWindow.AsSpan(span.Start - _characterWindowStartPositionInText, span.Length);
         }
 
         /// <summary>
-        /// Start parsing a new lexeme.
+        /// Moves this window to point at the given position in the text.  This will ensure that reading characters 
+        /// (and normally characters after it) will be fast.
         /// </summary>
-        public void Start()
-        {
-            _lexemeStart = _offset;
-        }
```

Comment:
no need "start" teh text window.  it does not care about lexemes.  it just cares about caching the most likely segment of chars needed by the lexer to read from.